src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,79 @@
+package duke.parser;
+
+import duke.command.Command;
+import duke.command.ByeCommand;
+import duke.command.DeleteCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.command.TaskCommand;
+
+import duke.exception.DukeNotTaskException;
+
+/**
+ * A Class that parse the command input.
+ *
+ * @author marioalvaro
+ */
+public class Parser {
+    /**
+     * Enum for all the possible command words
+     */
+    enum CommandWord {
+        BYE,
+        DELETE,
+        LIST,
+        MARK,
+        UNMARK,
+        TODO,
+        DEADLINE,
+        EVENT,
+        FIND,
+    }
+
+    /**
+     * Method that parse the command input into different Command
+     * @param input A String of command
+     * @return A Command object that correlate to the input
+     * @throws Exception thrown if the command is unknown
+     */
+    public static Command parse(String input) throws Exception {
+        try {
+            CommandWord commandWord = CommandWord.valueOf(input.split(" ")[0].toUpperCase());
+
+            String[] splitTask = input.split(" ", 2);
```

Comment:
I like the way you break a string into array of strings and use it among classes.